packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -467,6 +491,39 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
     return FlutterCommandResult.success();
   }
 
+  /// Adds a `platform` argument to the command.
+  ///
+  /// The type of the argument is [List]. The valid options are: `ios`, `android`, `windows`, `linux`, `macos`, `web`.
+  ///
+  /// These platforms should indicate what platforms the project will support after running the command.
+  /// The result can be used in generate methods such as [generateApp] and [generatePlugin].
+  ///
+  /// Adding argument is optional if the command does not require the user to explicitly state what platforms the project supports.
+  void _addPlatformsOptions() {
+    argParser.addMultiOption('platforms',
+        help: 'the platforms supported by this project.'
+          'This argument only works when the --template is set to app or plugin.'
+          'Respective folders will be generated in the target project.'
+          'When adding platforms to a plugin project, the pubspec.yaml will be updated with the requested platform.'
+          'When adding windows, linux or macos platforms, the platform feature has to be enable. Otherwise it would be a no-op.'
+          'To enable desktop features, run `flutter config --enable-<desktopName>-desktop.'
```

Comment:
Maybe: Adding desktop platforms requires the corresponding desktop config setting to be enabled